CTFd/admin.py

```diff
@@ -711,8 +697,14 @@ def admin_stats():
     wrong_count = db.session.query(db.func.count(WrongKeys.id)).first()[0]
     solve_count = db.session.query(db.func.count(Solves.id)).first()[0]
     challenge_count = db.session.query(db.func.count(Challenges.id)).first()[0]
-    most_solved_chal = Solves.query.add_columns(db.func.count(Solves.chalid).label('solves')).group_by(Solves.chalid).order_by('solves DESC').first()
-    least_solved_chal = Challenges.query.add_columns(db.func.count(Solves.chalid).label('solves')).outerjoin(Solves).group_by(Challenges.id).order_by('solves ASC').first()
+    most_solved_chal_sub = db.session.query(Solves.chalid, db.func.count(Solves.chalid).label('solves')).group_by(Solves.chalid).subquery()
+    most_solved_chal = Challenges.query.add_columns(coalesce(most_solved_chal_sub.columns.solves, 0).label('solves')) \
+        .join(most_solved_chal_sub, most_solved_chal_sub.columns.chalid == Challenges.id) \
+        .order_by(most_solved_chal_sub.columns.solves.desc().nullslast()).first()
+    least_solved_chal_sub = db.session.query(Solves.chalid, db.func.count(Solves.chalid).label('solves')).group_by(Solves.chalid).subquery()
+    least_solved_chal = Challenges.query.add_columns(coalesce(most_solved_chal_sub.columns.solves, 0).label('solves')) \
+        .outerjoin(most_solved_chal_sub, most_solved_chal_sub.columns.chalid == Challenges.id) \
+        .order_by(most_solved_chal_sub.columns.solves.nullsfirst()).first()
```

Comment:
I don't know if this is how sqlalchemy works, but it seems like if you use coalesce then you should be able to order by the result column ``solves`` since those will already be replaced with zeros.

I'd test it, but my connection to our server just died - looks like a random power failure or an issue with our modem - so it'll have to wait until I can get in and debug what's going on with our server.